### PR TITLE
Check for error instances

### PR DESCRIPTION
**Pull request by danmilon**
Picking up from the discussion at #6 (took me some time!)

Checks if `attrs.error` is an Error instance, and replaces it with the stacktrace as a string or the message if there is no stack attached.

Also there's a test case.

### CODE REVIEW

File: test/jog.js [hunk 1]
```diff
@@ -30,6 +30,21 @@ describe('Jog', function(){
...
+      var store = {
+        add: function(obj){
+          obj.error.should.equal(err.stack);
+          obj.x.should.equal('y');
+          done();
+        }
+      };
+
+      var log = new Jog(store);
+      log.error('something happened', { error: err, x: 'y' });
```
Comment (tj) [inline test/jog.js]: might be nice to also support `log.error('request', err)` directly, expanding to `{ message: m, stack: s }`, both would be nice then you could just `--type error --map message` and get a nice list

Comment (danmilon) [inline test/jog.js]: How would `--type error` work with `log.error('request', err)`? (type equals request)

Comment (tj) [inline test/jog.js]: oops sorry `--level error`

**Comment by danmilon** [comment]
And you can do `-s error --map error.message`

**Pull request merged**